node/src/utils/fmt_limit.rs

```diff
@@ -0,0 +1,62 @@
+//! Wrappers to display a limited amount of data from collections using `fmt`.
+
+use std::fmt::{self, Debug, Formatter};
+
+/// A display wrapper showing a limited amount of items in a slice.
+pub(crate) struct LimitSlice<'a, T, const N: usize = 10>(&'a [T]);
+
+impl<'a, T, const N: usize> LimitSlice<'a, T, N> {
+    /// Creates a new limited slice instance.
+    pub(crate) fn new(slice: &'a [T]) -> Self {
+        LimitSlice(slice)
+    }
+}
+
+impl<'a, T, const N: usize> Debug for LimitSlice<'a, T, N>
+where
+    T: Debug,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str("[")?;
+
+        let count = N.min(self.0.len());
+        for (idx, item) in self.0.iter().take(count).enumerate() {
+            Debug::fmt(item, f)?;
+            if idx + 1 != count {
+                f.write_str(", ")?;
+            }
+        }
+
+        if count != self.0.len() {
+            f.write_str(" ...")?;
+        }
+
+        f.write_str("]")?;
+        Ok(())
+    }
+}
+
+// Note: If required, a `Display` implementation can be added easily for `LimitSlice`.
+
+#[cfg(test)]
+mod tests {
+    use crate::utils::fmt_limit::LimitSlice;
+
+    #[test]
+    fn limit_debug_works() {
+        let collections: Vec<_> = (0..5).into_iter().collect();
+
+        // Sanity check.
+        assert_eq!(format!("{:?}", collections), "[0, 1, 2, 3, 4]");
+
+        assert_eq!(
+            format!("{:?}", LimitSlice::<'_, _, 3>::new(collections.as_slice())),
+            "[0, 1, 2 ...]"
```

Comment:
Why is there no comma after `2` and before `...` ? `2` would have an `idx` of 2 (`enumerate` starts from 0) so `idx + 1 != count` b/c `2 + 1 != 5`

Reply:
> Why is there no comma after 2 and before ... ?

Because I thought it'd look nicer that way :cry: . I guess `[0, 1, 2, ...]` is still better, now that I think about it...

Reply:
Ah, right... This particular example uses different `N` . For some reason I thought it would be `5`.

Reply:
I changed the format and added more tests.